### PR TITLE
Better error for missing index in CRV2

### PR DESCRIPTION
**Pull request by gupnik**
Fixes https://github.com/paritytech/polkadot-sdk/issues/4552

### CODE REVIEW

**Comment by gupnik** [comment]
bot fmt

File: substrate/frame/support/procedural/src/runtime/parse/mod.rs [hunk 1]
```diff
@@ -187,6 +187,13 @@ impl Def {
 				}
 			}
 
+			if pallet_index.is_none() {
```
Comment (bkchr) [inline substrate/frame/support/procedural/src/runtime/parse/mod.rs:190]:
```suggestion
			let Some(pallet_index) = pallet_index else {
```

File: substrate/frame/support/procedural/src/runtime/parse/mod.rs [hunk 2]
```diff
@@ -187,6 +187,13 @@ impl Def {
...
+				if let syn::Item::Type(item) = item {
```
Comment (bkchr) [inline substrate/frame/support/procedural/src/runtime/parse/mod.rs:191]: This if doesn't make sense? I mean the index can only be set when the item is a type, at least judging based on the code above.

**Review by bkchr — APPROVED** [review]

File: substrate/frame/support/procedural/src/runtime/parse/mod.rs (hunk 2 above)
Comment (gupnik) [reply substrate/frame/support/procedural/src/runtime/parse/mod.rs:191]: > This if doesn't make sense? I mean the index can only be set when the item is a type, at least judging based on the code above.

So, the module marked with #[frame_support::runtime] can have arbitrary items. For example,
```rust
#[frame_support::runtime]
mod runtime {
    use super::*;
    
    pub type System = frame_system;
}
```

In this case, you only want this check for `pub type System = frame_system;` and not for `use super::*;`. 

Basically, this change requires all type aliases within the runtime macro to be annotated with `#[runtime::pallet_index]` while the other items are preserved.

File: substrate/frame/support/procedural/src/runtime/parse/mod.rs (hunk 1 above)
Comment (gupnik) [reply substrate/frame/support/procedural/src/runtime/parse/mod.rs:190]: `pallet_index` can be None for all items other than type aliases.

**Pull request merged**